fix(posts): introduce responsive components to embed iframes

elisa-a-v opened the pull request:
To make embedding content easier, two new components have been added: `ResponsiveIFrame` and `YouTubeVideo`.

YouTube videos require a specific URL to be embedded in iframes, which is why using the standard URL doesn't work, but the new component takes care of this by attempting to tweak the URL, allowing us to simply do:

```html
<YouTubeVideo src="https://www.youtube.com/watch?v=hEXl4rjlN7g"/>
```

However, the embed URL still needs to be used if we want the video to start in a specific time. Instead of copying the entire content provided by YouTube in Share > Embed, we just need to copy what's in `src`:

<img width="530" height="486" alt="image" src="https://github.com/user-attachments/assets/4a5d9839-9457-4aa5-9454-29ee6f7ff95b" />

Then:

```html
<YouTubeVideo src="https://www.youtube.com/embed/hEXl4rjlN7g?si=5DlwtXkrKFfJaESJ&amp;start=1045"/>
```

## Screenshots

<details>
<summary>Visualizations before / after</summary>

<img width="300" alt="Screenshot from 2025-07-31 20-11-58" src="https://github.com/user-attachments/assets/25214387-5369-4cd0-9ccc-b9996b4644f0" />
<img width="300" alt="Screenshot from 2025-07-31 20-11-44" src="https://github.com/user-attachments/assets/1478ecbd-7905-4788-bb8e-c7eb8a75f638" />

</details>

<details>
<summary>Buttons layout before / after</summary>

<img width="300" alt="Screenshot from 2025-07-31 17-55-22" src="https://github.com/user-attachments/assets/46a1a40e-c327-43ef-bd48-d40db0d1e780" />

<img width="300" alt="Screenshot from 2025-07-31 17-55-11" src="https://github.com/user-attachments/assets/397b37dc-d3a2-4254-bb54-6ead331f1aab" />

</details>

<details>
<summary>YouTube Videos before / after</summary>

<img width="300" alt="Screenshot from 2025-07-31 17-57-14" src="https://github.com/user-attachments/assets/4afc02c4-6919-46c2-9030-88d2257e12e9" />
<img width="300" alt="Screenshot from 2025-07-31 17-57-20" src="https://github.com/user-attachments/assets/4efeeb3d-b63e-4dc7-b1ea-f8ec384d20a9" />

</details>

mlissner reviewed (changes requested):
One thing that'd be nice is for this to be added to the example.mdx file. Otherwise, looks great!

elisa-a-v commented:
Great idea! I've updated the file to include both `YouTubeVideo` and `ResponsiveIFrame`, and I added a couple of minor tweaks to that page [here](https://github.com/freelawproject/free.law/pull/308/commits/7f6271f58fff2b48485f72d66668bd09cf47cdb5):
- The buttons in that page were also stacking on top of each other in smaller screens so that's been fixed.
- The `img-responsive` class isn't declared anywhere, and inspecting in the browser it isn't recognized either so I just removed it.
- Added a link to the CourtListener markdown help page.

What do you think?
